modules/core/src/main/java/org/apache/ignite/internal/IgnitionEx.java

```diff
@@ -3209,6 +3214,13 @@ public void setCounter(int cnt) {
                 this.cnt = cnt;
             }
         }
+
+        /**
+         *
+         * @return whether the startLatch has been counted down, thereby indicating that the kernal has full started.
+         */
+        public boolean hasStartLatchCompleted() { return startLatch.getCount() == 0; }
```

Comment:
Every please unroll method body to 3 lines. You can import code conventions in your IDEA from https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines

Can you also discuss this approach on the developers list? I'm not sure we should use this latch for checking.